lib/AST/ConformanceLookupTable.cpp

```diff
@@ -818,6 +818,8 @@ ConformanceLookupTable::getConformance(NominalTypeDecl *nominal,
     // declared.
     auto *conformingClass = cast<ClassDecl>(conformingNominal);
     Type superclassTy = type->getSuperclassForDecl(conformingClass);
```

Comment:
I still don’t get how getSuperclassForDecl() can ever fail. What is ‘type’?

Reply:
`getSuperclassForDecl()` can fail if the superclass-type-as-written is ill-formed.

Reply:
So getSuperclassDecl() can succeed while getSuperclass() can fail if you found the decl but the type is otherwise bad? That makes sense I guess.

Reply:
Yes. It's a new failure state because `getSuperclassDecl()` is lazier (checks less) than `getSuperclass()`.